### PR TITLE
[#89729346] Fixes previewParser option for rails projects.

### PR DESCRIPTION
**Pull request by wilatkabam**
https://www.pivotaltracker.com/story/show/89729346

### CODE REVIEW

File: markitup/jquery.markitup.js [hunk 1]
```diff
@@ -546,9 +546,10 @@
 					options.previewHandler(parsedData);
 				} else if (options.previewParserPath !== '') {
 					$.ajax({
-						type: options.previewParserAjaxType,
-						dataType: 'text',
-						global: false,
+						method: options.previewParserAjaxMethod,
+						beforeSend: function(xhr) {
+							xhr.setRequestHeader('X-CSRF-Token', $('meta[name="csrf-token"]').attr('content'));
```
Comment (jaygeeseman) [inline markitup/jquery.markitup.js]: This should be passed in as it's not part of marktup. You can pass a function definition in the settings hash and use it for the callback.

Comment (wilatkabam) [inline markitup/jquery.markitup.js]: I'm not sure how we can access internal functions from a callback we set during startup. The options can be replaced by string literals, but inside of the success function below - how would we access the writeInPreview function.

Comment (wilatkabam) [inline markitup/jquery.markitup.js]: Maybe a good compromise would be to create a way to extend the object used by Ajax.
We can add a new option to the setup object:
`self.settings = {
      ...
      previewParserAjaxObj: {
        beforeSend: function(xhr) {
          xhr.setRequestHeader('X-CSRF-Token', $('meta[name="csrf-token"]').attr('content'));
        }
      }
}`

And then extend the object used in Ajax with whatever is user defined:

`var ajaxSettings = { 
  method: options.previewParserAjaxMethod,
  url: options.previewParserPath,
  data: options.previewParserVar+'='+encodeURIComponent(parsedData),
  success: function(data) {
    writeInPreview( localize(data, 1) ); 
  }
};
$.extend(ajaxSettings, options.previewParserAjaxObj);
$.ajax(ajaxSettings)`

Comment (jaygeeseman) [inline markitup/jquery.markitup.js]: The success callback is part of markitup, correct? It shouldn't be a factor? I'm not suggesting putting all callbacks in the settings hash.

**Comment by wilatkabam** [comment]
@jaygeeseman I ended not doing the extend inline so that we can do some minor error checks to make sure that the extend call won't break anything.

File: markitup/jquery.markitup.js [hunk 2]
```diff
@@ -545,16 +546,20 @@
...
+							data: options.previewParserVar+'='+encodeURIComponent(parsedData),
+							success: function(data) {
+								writeInPreview( localize(data, 1) ); 
+							}
+						};
+						if(options.previewParserAjaxSettings && typeof options.previewParserAjaxSettings === 'object'){
+							$.extend(ajaxSettings, options.previewParserAjaxSettings);
 						}
-					});
+						$.ajax(ajaxSettings);
```
Comment (jaygeeseman) [inline markitup/jquery.markitup.js]: Indentation is off. Looks good otherwise.

**Pull request merged**